### PR TITLE
Fix installation doc

### LINKED ISSUE #6120: Sphinx build package requirements are incomplete
The listed [requirements for building the sphinx docs ](http://astropy.readthedocs.io/en/latest/install.html#building-documentation) are incomplete:  

- Sphinx (and its dependencies) 1.0 or later
- Graphviz
- Astropy-helpers (Astropy and most affiliated packages include this as a submodule in the source repository, so it does not need to be installed separately.)
- Pillow
- (optional) sphinx-gallery

At the very least one also needs PyYaml and presumably matplotlib (for the plot directives), but maybe more?  Here is `pip-requirements-doc`:
```
Cython>=0.21
jinja2
pyyaml
scipy
pandas
h5py
beautifulsoup4
bleach
jplephem

# below here are used only for tests
skyfield
mpmath
```
This does not include matplotlib which I find confusing, but anyway, someone who understands the doc building process in detail could probably update the list of requirements in the sphinx docs.

### PR DESCRIPTION
**Pull request by pllim**
Fix #6120 -- Is this clearer?

Also added a note about #7818.

### CODE REVIEW

**Comment by keflavich** [comment]
:+1: this is a good stopgap until I can put some real time into tests

File: pip-requirements [hunk 1]
```diff
@@ -1,2 +1,2 @@
-numpy>=1.10.0,<1.15.3
+numpy>=1.13.0,<1.15.3
```
Comment (bsipocz) [inline pip-requirements]: wow, good find.

File: docs/install.rst [hunk 2]
```diff
@@ -2,17 +2,21 @@
...
 Requirements
 ============
 
 Astropy has the following strict requirements:
 
-- `Python <https://www.python.org/>`_ 3.5 or 3.6
+- `Python <https://www.python.org/>`_ 3.5 or later
 
-- `Numpy`_ |minimum_numpy_version| or later
+- `Numpy`_ |minimum_numpy_version| or later (but not 1.15.3)
```
Comment (astrofrog) [inline docs/install.rst]: I wonder if this is too specific - after all *most* of astropy does work fine with this version, and there may be other bug fix releases of numpy where some tests fail.

Comment (pllim) [reply docs/install.rst]: It is not even the tests. RTD could not build without the `!=` pinning. I am open to suggestions though. https://readthedocs.org/projects/astropy/builds/8024893/

**Comment by bsipocz** [comment]
Merging this one, too. Surely we can do more polishing for this page, but small steps are better than waiting for the best solution.

**Comment by bsipocz** [comment]
Thank you @pllim!

File: docs/install.rst (hunk 2 above)
Comment (bsipocz) [reply docs/install.rst]: Currently this is useful to have as people started to open issues they run into using astropy with 1.15.3. 

When the new numpy release is out, I would do agree with @astrofrog here that this is too specific (we had issues with other minor bugfix versions before, and never listed them once newer releases became available.)

**Pull request merged**